File: app/src/modules/settings/api/data-transfer-objects/update-service.ts
import { ServiceConfiguration } from '@/modules/settings/api/entities/service-configuration';
import { VehicleCategory } from '@/modules/app/api';
import { ServicePricingMethod } from '@/modules/settings/api/value-objects/service-pricing-method';

export type UpdateService = {
    id: string;
    name: string;
    description?: string;
    pricingMethod: ServicePricingMethod;
    configurations: { vehicleCategoryId: string | null; price: number; duration: number }[];
};

File: api/src/Domain/Settings/DataTransferObjects/UpdateService.cs
using System.Collections.Generic;

namespace DetailingArsenal.Domain.Settings {
    public class UpdateService : IDataTransferObject {
        public string Name { get; }
        public string? Description { get; }
        public ServicePricingMethod PricingMethod { get; }
        public List<UpdateServiceConfiguration> Configurations { get; }

        public UpdateService(string name, string? description, ServicePricingMethod pricingMethod, List<UpdateServiceConfiguration> configurations) {
            Name = name;
            Description = description;
            PricingMethod = pricingMethod;
            Configurations = configurations;
        }
    }
}